src/trace/trace.cpp

```diff
@@ -286,7 +339,7 @@ otf2::definition::metric_member Trace::metric_member(const std::string& name,
                                                      otf2::common::type value_type,
                                                      const std::string& unit)
 {
-    auto ref = metric_members_.size();
+    auto ref = metric_member_ref();
```

Comment:
with these `*_ref()` methods now in place, I'd rather remove this local variable.